artifactory/commands/repository/repository.go

```diff
@@ -1017,6 +1019,336 @@ func remoteGenericHandler(servicesManager artifactory.ArtifactoryServicesManager
 	return err
 }
 
+var federatedRepoHandlers = map[string]repoHandler{
+	Maven:     federatedMavenHandler,
+	Gradle:    federatedGradleHandler,
+	Ivy:       federatedIvyHandles,
+	Sbt:       federatedSbtHandler,
+	Helm:      federatedHelmHandler,
+	Cocoapods: federatedCocoapodsHandler,
+	Opkg:      federatedOpkgHandler,
+	Rpm:       federatedRpmHandler,
+	Nuget:     federatedNugetHandler,
+	Cran:      federatedCranHandler,
+	Gems:      federatedGemsHandler,
+	Npm:       federatedNpmHandler,
+	Bower:     federatedBowerHandler,
+	Debian:    federatedDebianHandler,
+	Composer:  federatedComposerHandler,
+	Pypi:      federatedPypiHandler,
+	Docker:    federatedDockerHandler,
+	Vagrant:   federatedVagrantHandler,
+	Gitlfs:    federatedGitlfsHandler,
+	Go:        federatedGoHandler,
+	Conan:     federatedConanHandler,
+	Chef:      federatedChefHandler,
+	Puppet:    federatedPuppetHandler,
+	Alpine:    federatedAlpineHandler,
+	Generic:   federatedGenericHandler,
+}
+
+func federatedMavenHandler(servicesManager artifactory.ArtifactoryServicesManager, jsonConfig []byte, isUpdate bool) error {
+	params := services.NewMavenFederatedRepositoryParams()
+	err := json.Unmarshal(jsonConfig, &params)
+	if errorutils.CheckError(err) != nil {
+		return err
+	}
```

Comment:
Let's take this opportunity and make the following simplification all accross this file - 
```
	if err := json.Unmarshal(jsonConfig, &params); err != nil {
		return errorutils.CheckError(err)
	}
```

Reply:
if I simplify like this, err variable will not be defined out of the if condition, and we use it in the below lines, we can change it like this : 

	var err error
	if err = json.Unmarshal(jsonConfig, &params); err != nil {
		return errorutils.CheckError(err)
	}